### PR TITLE
update session-creation logic

### PR DESCRIPTION
**Pull request by brycahta**
*Issue #, if available:* N/A

*Description of changes:*
* Update aws session creation logic to align with Instance-Selector
* Pass region from CLI to agent
* If users do not provide a region, then do **not** assign in userConfig

*[Testing](https://youtu.be/mqFJ_oaG4nw?t=3):*
* `make clean && make test` locally

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/config/config_test.go [hunk 1]
```diff
@@ -111,7 +111,7 @@ func TestPopulateTestFixtureForResumedRun(t *testing.T) {
 
 func TestParseCliArgsEnvSuccess(t *testing.T) {
 	resetFlagsForTest()
-	os.Setenv("AWS_DEFAULT_REGION", "AWS_DEFAULT_REGION")
+	os.Setenv("AWS_REGION", "us-weast-1")
```
Comment (brycahta) [inline pkg/config/config_test.go:114]: not a [typo](https://youtu.be/RuJNUXT2a9U?t=9)

File: go.mod [hunk 2]
```diff
@@ -4,5 +4,7 @@ go 1.14
 
 require (
 	github.com/aws/aws-sdk-go v1.31.5
+	github.com/mitchellh/go-homedir v1.1.0
```
Comment (bwagner5) [inline go.mod:7]: can you add these new pkgs to the third party licenses file

**Review by bwagner5 — CHANGES_REQUESTED** [review]
looks good, just need attributions

File: go.mod (hunk 2 above)
Comment (brycahta) [reply go.mod:7]: good catch!

**Review by bwagner5 — APPROVED** [review]
:+1:

**Pull request merged**